lsp/core/aws-lsp-core/src/content/schemaProviderBuilder.ts

```diff
@@ -0,0 +1,34 @@
+import { URI } from 'vscode-uri'
+import { ContentRequestMiddleware, ContentRequestMiddlewareDelegate } from './contentRequestMiddleware'
+import { SchemaProvider } from './schemaProvider'
+
+/**
+ * Creates a middleware pipeline for resolving requests for a schema.
+ * Middleware handlers will be called in the order they are added, until a
+ * handler returns a response. At this point, the handlers in the call stack
+ * have a chance to handle/process the response.
+ */
+export class SchemaProviderBuilder {
+    private readonly handlers: ContentRequestMiddleware[] = []
+
+    public addHandler(handler: ContentRequestMiddleware): SchemaProviderBuilder {
+        this.handlers.push(handler)
+        return this
+    }
+
+    public build(): SchemaProvider {
```

Comment:
A pipeline gives us more flexibility. For example, we can have a handler that uses a cache if the content is considered up to date, otherwise delegates. Then we can have another handler that obtains (from http, or elsewhere). Both of these could be used for the same scheme. Rather than having one large implementation, we can separate these concerns, and even use them in different orchestrations. Peek ahead to #506 for an example of this.